src/main/java/com/bobocode/svydovets/beans/scanner/ComponentBeanScanner.java

```diff
@@ -33,4 +40,69 @@ public Map<String, BeanDefinition> scan(String packageName) {
                         beanDefinition -> beanDefinition
                 ));
     }
+
+    @Override
+    public void fillDependsOn(Map<String, BeanDefinition> beanDefinitions) {
+        for (BeanDefinition beanDefinition : beanDefinitions.values()) {
+            Class<?> beanClass = beanDefinition.getBeanClass();
+            if (beanClass == null) {
+                continue;
+            }
+
+            beanDefinition.setDependsOn(findDependsOn(beanDefinitions, beanClass));
+        }
+    }
+
+    private static String[] findDependsOn(Map<String, BeanDefinition> beanDefinitions, Class<?> beanClass) {
+        var dependsOn = new ArrayList<>();
+        for (Field field : beanClass.getDeclaredFields()) {
+            if (!field.isAnnotationPresent(Inject.class)) {
+                continue;
+            }
+
+            String beanName = field.getAnnotation(Inject.class).value();
+            if (StringUtils.isNotEmpty(beanName)) {
+                findDepnesOnByName(beanDefinitions, dependsOn, beanName);
+            } else {
+                findDependsOnByClass(beanDefinitions, beanClass, dependsOn, field);
+            }
+        }
+
```

Comment:
can be changed too
`  Arrays.stream(beanClass.getDeclaredFields())
          .filter(field -> field.isAnnotationPresent(Inject.class))
          .forEach(field -> {
              String beanName = field.getAnnotation(Inject.class).value();
              if (StringUtils.isNotEmpty(beanName)) {
                  findDepnesOnByName(beanDefinitions, dependsOn, beanName);
              } else {
                  findDependsOnByClass(beanDefinitions, beanClass, dependsOn, field);
              }
          });`